File: Results/GeneLists/format_gene_lists.py

# Import
import os
import sys

# Input
geneListSet = [ "/hpcwork/izkf/projects/malmo/GeneLists/all_genes_raw.txt",
                "/hpcwork/izkf/projects/malmo/GeneLists/coexpressed_genes_raw.txt"
]

# Dictionary of symbols
# Uppercase to Uppercase
# These mappings were made using uniprot and genecards
symbolDict = dict([
    # SH
    ("C9ORF150","LURAP1L"),
    ("FAM123C","AMER3"),
    ("FAM148A","C2CD4A"),
    ("KIAA0774","MTUS2"),
    ("KIAA1383","MAP10"),
    ("KIAA1486","NYAP2"),
    ("BRAC","T"),
    ("ODZ3","TENM3"),
    ("BAPX1","NKX3-2"),
    ("CEBP","CPEB1"),
    ("DELTAEF1","ZEB1,ZEB2"),
    # EG
    ("AAA1","SLC7A10"),
    ("C15ORF23","KNSTRN"),
    ("C20ORF54","SLC52A3"),
    ("C6ORF162","SMIM8"),
    ("D4S234E","NSG1"),
    ("EVI1","MECOM"),
    ("FLJ25770","CCDC158"),
    ("GFI","GFI1,GFI1B"),
    ("MS4A8B","MS4A8"),
    ("PBX","PBX1,PBX2,PBX3,PBX4"),
    ("RORA2","RORA"),
    ("TCF1","HNF1A,TCF7"),
    ("TCFE2A","TCF3"),
    ("TEAD","TEAD1,TEAD2,TEAD3,TEAD4,TEAD5"),
    ("ZNF42","MZF1")
])

# Iterate on gene list
for geneListFileName in geneListSet:

    # Read gene list
    geneList = [] # In the end it will be RAW, UPPER, FINAL
    geneListFile = open(geneListFileName,"r")
    for line in geneListFile:
        geneList.append([line.strip(),line.strip().upper()])
    geneListFile.close()

    # Insert correct symbols
    for e in geneList:
        try:
            e.append(symbolDict[e[1]])
        except Exception:
            e.append(e[1])

    # Writing mapping file
    prefixFileName = geneListFileName[:-7]
    outputFile = open(prefixFileName+"mapping.txt","w")
    for e in geneList:
        for k in e[2].split(","):
            outputFile.write("\t".join([e[0],k])+"\n")
    outputFile.close()

    # Writing format file
    outputFile = open(prefixFileName+"preformat.txt","w")
    for e in geneList:
        outputFile.write(e[1]+"\n")
        for k in e[2].split(","):
            outputFile.write(k+"\n")
    outputFile.close()

    # Remove duplicates from format file
    os.system("sort "+prefixFileName+"preformat.txt | uniq > "+prefixFileName+"format.txt")
    os.system("rm "+prefixFileName+"preformat.txt")

File: Results/GeneLists/format_gene_lists.R

candidate_genes = read.table("candidate_genes.txt", sep = "\t", header = F)
#assoc_file = read.table("association_file.bed", sep = "\t", header = F)

candidate_genes$V1 = toupper(candidate_genes$V1)
#not_in_assoc_file = setdiff(candidate_genes$V1, toupper(assoc_file$V4))

candidate_genes$V1[candidate_genes$V1 == "C9ORF150"] = "LURAP1L"
candidate_genes$V1[candidate_genes$V1 == "FAM123C"] = "AMER3"
candidate_genes$V1[candidate_genes$V1 == "FAM148A"] = "C2CD4A"
candidate_genes$V1[candidate_genes$V1 == "KIAA0774"] = "MTUS2"
candidate_genes$V1[candidate_genes$V1 == "KIAA1383"] = "MAP10"
candidate_genes$V1[candidate_genes$V1 == "KIAA1486"] = "NYAP2"
candidate_genes$V1[candidate_genes$V1 == "BRAC"] = "T"
candidate_genes$V1[candidate_genes$V1 == "ODZ3"] = "TENM3"
candidate_genes$V1[candidate_genes$V1 == "BAPX1"] = "NKX3-2"
candidate_genes$V1[candidate_genes$V1 == "CEBP"] = "CPEB1"
candidate_genes$V1[candidate_genes$V1 == "DELTAEF1"] = "ZEB2"

#intersect(candidate_genes$V1, toupper(assoc_file$V4))

write.table(toupper(candidate_genes$V1),file = "candidate_genes_format.txt", sep = "\t", col.names = F, row.names = F, quote = F)

######################################################################################

candidate_genes = read.table("co_expressed_genes.txt", sep = "\t", header = F)
#assoc_file = read.table("association_file.bed", sep = "\t", header = F)

candidate_genes$V1 = toupper(candidate_genes$V1)
#not_in_assoc_file = setdiff(candidate_genes$V1, toupper(assoc_file$V4))

candidate_genes$V1[candidate_genes$V1 == "C9ORF150"] = "LURAP1L"

#intersect(candidate_genes$V1, toupper(assoc_file$V4))

write.table(toupper(candidate_genes$V1),file = "co_expressed_genes_format.txt", sep = "\t", col.names = F, row.names = F, quote = F)





